src/tracer/MOM_offline_aux.F90

```diff
@@ -101,66 +101,63 @@ subroutine update_h_vertical_flux(G, GV, ea, eb, h_pre, h_new)
   ! Update h_new with convergence of vertical mass transports
   do j=js-1,je+1
     do i=is-1,ie+1
-
       ! Top layer
-      h_new(i,j,1) = max(0.0, h_pre(i,j,1) + (eb(i,j,1) - ea(i,j,2) + ea(i,j,1) ))
-      h_new(i,j,1) = h_new(i,j,1) + &
-          max(0.0, 1.0e-13*h_new(i,j,1) - h_pre(i,j,1))
+      h_new(i,j,1) = max(0.0, h_pre(i,j,1) + ((eb(i,j,1) - ea(i,j,2)) + ea(i,j,1)))
+      h_new(i,j,1) = h_new(i,j,1) + max(0.0, 1.0e-13*h_new(i,j,1) - h_pre(i,j,1))
 
       ! Bottom layer
-!        h_new(i,j,nz) = h_pre(i,j,nz) + (ea(i,j,nz) - eb(i,j,nz-1)+eb(i,j,nz))
-      h_new(i,j,nz) = max(0.0, h_pre(i,j,nz) + (ea(i,j,nz) - eb(i,j,nz-1)+eb(i,j,nz)))
-      h_new(i,j,nz) = h_new(i,j,nz) + &
-          max(0.0, 1.0e-13*h_new(i,j,nz) - h_pre(i,j,nz))
-
+      h_new(i,j,nz) = max(0.0, h_pre(i,j,nz) + ((ea(i,j,nz) - eb(i,j,nz-1)) + eb(i,j,nz)))
+      h_new(i,j,nz) = h_new(i,j,nz) + max(0.0, 1.0e-13*h_new(i,j,nz) - h_pre(i,j,nz))
     enddo
 
     ! Interior layers
     do k=2,nz-1 ; do i=is-1,ie+1
-
       h_new(i,j,k) = max(0.0, h_pre(i,j,k) + ((ea(i,j,k) - eb(i,j,k-1)) + &
-          (eb(i,j,k) - ea(i,j,k+1))))
-      h_new(i,j,k) = h_new(i,j,k) + &
-        max(0.0, 1.0e-13*h_new(i,j,k) - h_pre(i,j,k))
-
+                                              (eb(i,j,k) - ea(i,j,k+1))))
+      h_new(i,j,k) = h_new(i,j,k) + max(0.0, 1.0e-13*h_new(i,j,k) - h_pre(i,j,k))
```

Comment:
The 1.e-13 should be replaced with a GV%Angstrom here

Reply:
As I look at the code, I am confused about what was intended here.  The 1.0e-13*h seems to have been copied from the tracer advection code, where it added mass to a copy of the _previous_ thickness so that after advection, the updated thickness seen by the tracers will not less than 13 orders of magnitude smaller than the previous thickness.  In the case of advect_tracer(), this avoids truncation errors leading to crazy tracer concentrations in newly massless layers.  The actual model thicknesses are not impacted - only a local copy in the tracer advection code is - so that mass is still conserved even if tracers are perfectly conserved with advectively drying layers.

In the MOM_offline_aux.F90 code, is the _updated_ thickness that is being incremented if the old concentration was very thin. In other words, this impacts **_wetting_**, and not drying as before. I have no idea why we would need to do something special in this case. @ashao, do you recall why this was added, and if not, do you agree with me that this extra step of adding a little bit of mass to the new state can probably just be dropped, but with GV%Angstrom replacing 0.0 in the max on line 115 of MOM_offline_aux?

Reply:
I have commented out the lines that were the topic of this comment, as I do not think that they are either necessary or helpful.  However, I have retained them for now in case we do encounter a situation where they could be beneficial.  They could probably be removed altogether in a future commit.

Reply:
I agree with the change that you made. I'm pretty sure this was just a blind copy and paste to solve a problem that we won't encounter in the offline code.